src/util.js

```diff
@@ -57,32 +59,57 @@
     return false;
   }
 
-
   /*
-   * Get iterator of object which returns a { done: false, value: expression }
-   *   if it has not reached the end of the object, or otherwise it returns
+   * Gets an iterator for an object.
+   *
+   * An iterator returns { done: false, value: expression } if it has not
+   *   reached the end of the array-like object, or otherwise it returns
    *   { done: true, value: undefined }.
-   * @param {!Object} object: The object to iterate over.
-   * @param {Function} callback: Defines the value attribute for the iterator.
-   *   The function takes two arguments (key, value), and should return one of
-   *   the following: key, value, or an array containing one or more of
-   *   (key, value). Generally, key is used for the keys() method, value is
-   *   used for the values() method, and [key, value] for the entries() method.
+   * See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols
+   * for more information about iterators.
+   * @param {!Array} keys: An array of keys to iterate over.
+   * @param {Function} callback: A function that returns the value for the
+   *   iterator. The function takes a single argument: the current key and
+   *   should return one of the following: key, value, or an array containing
+   *   one or more of (key, value). Generally, key is used for the keys()
+   *   method, value is used for the values() method, and [key, value] for
+   *   the entries() method.
    */
-  function iterator(object, callback) {
-    return {
-      index: 0,
+  function iterator(keys, callback) {
+    var index = 0;
+    var result = {
       next: function() {
-        if (this.index < object.length) {
-          return { done: false, value: callback(this.index, object[this.index++]) };
-        } else return { done: true, value: undefined };
+        if (index < keys.length) {
+          var key = keys[index++];
+          return { done: false, value: callback(key) };
+        } else {
+          return { done: true, value: undefined };
+        }
       }
-    }
+    };
+    result[Symbol.iterator] = function() { return result; };
+    return result;
```

Comment:
small nit: you could hide `next` rather than exposing it, if `Symbol.iterator` is the intended use